Entity/Context/DocumentImportContext.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+
+namespace EMS\CoreBundle\Entity\Context;
+
+
+use Doctrine\ORM\EntityManager;
+use EMS\CoreBundle\Entity\ContentType;
+use EMS\CoreBundle\Entity\Environment;
+use EMS\CoreBundle\Repository\ContentTypeRepository;
+use EMS\CoreBundle\Repository\RevisionRepository;
+
+class DocumentImportContext
+{
+
+    /** @var ContentType */
+    private $contentType;
+    /** @var string */
+    private $contentTypeName;
+    /** @var string */
+    private $lockUser;
+    /** @var bool */
+    private $rawImport;
+    /** @var EntityManager */
+    private $entityManager;
+    /** @var RevisionRepository */
+    private $revisionRepository;
+    /** @var bool */
+    private $indexInDefaultEnv;
+    /** @var bool */
+    private $signData;
+    /** @var ContentTypeRepository */
+    private $contentTypeRepository;
+    /** @var Environment */
+    private $environment;
+    /** @var bool */
+    private $finalize;
+    /** @var bool */
+    private $force;
+
+    public function __construct(EntityManager $entityManager, string $contentTypeName, string $lockUser, bool $rawImport, bool $signData, bool $indexInDefaultEnv, bool $finalize, bool $force)
+    {
+        $this->contentTypeName = $contentTypeName;
+        $this->indexInDefaultEnv = $indexInDefaultEnv;
+        $this->signData = $signData;
+        $this->lockUser = $lockUser;
+        $this->rawImport = $rawImport;
+        $this->finalize = $finalize;
+        $this->entityManager = $entityManager;
+        $this->force = $force;
+
+        $this->entityManager->getConnection()->getConfiguration()->setSQLLogger(null);
+
+        $repository = $this->entityManager->getRepository('EMSCoreBundle:Revision');
+        if (! $repository instanceof RevisionRepository) {
+            throw new \Exception('Can not get the RevisionRepository');
+        }
+        $this->revisionRepository = $repository;
+
+        $repository = $this->entityManager->getRepository('EMSCoreBundle:ContentType');
+        if (! $repository instanceof ContentTypeRepository) {
+            throw new \Exception('Can not get the ContentTypeRepository');
+        }
+        $this->contentTypeRepository = $repository;
+
+        $contentType = $this->contentTypeRepository->findOneBy(array("name" => $this->contentTypeName, 'deleted' => false));
+        if (! $contentType instanceof ContentType) {
+            throw new \Exception(sprintf('Content type %s not found', $this->contentTypeName));
+        }
```

Comment:
Should not be part of the context (state)

Reply:
adcc14722bbdaf62cb0e3166cba8580d2dbaa827